compiler/src/dotty/tools/dotc/ast/Trees.scala

```diff
@@ -337,13 +337,16 @@ object Trees {
      *  a calling chain from `viewExists`), in that case the return position is NoSpan.
      *  Overridden in Bind
      */
-    def nameSpan: Span =
+    def nameSpan(using Context): Span =
       if (span.exists) {
         val point = span.point
-        if (rawMods.is(Synthetic) || name.toTermName == nme.ERROR) Span(point)
+        if (rawMods.is(Synthetic) || span.isSynthetic || name.toTermName == nme.ERROR) Span(point)
         else {
           val realName = name.stripModuleClassSuffix.lastPart
-          Span(point, point + realName.length, point)
+          var length = realName.length
+          if symbol.is(ExtensionMethod) && name.isExtensionName then
```

Comment:
Does this work on untyped trees?

Reply:
i've added a case to check the mods as well